pkg/storage/util/util.go

```diff
@@ -50,18 +53,46 @@ func UpdateCondition(cr *imageregistryv1.Config, conditionType string, status op
 	cr.Status.Conditions = conditions
 }
 
-func GetClusterVersionConfig(kubeconfig *rest.Config) (*configv1.ClusterVersion, error) {
-	client, err := configv1client.NewForConfig(kubeconfig)
+func GetInstallConfig(listers *regopclient.Listers) (*installer.InstallConfig, error) {
+	cm, err := listers.InstallerSecrets.Get("cluster-config-v1")
 	if err != nil {
 		return nil, err
 	}
-	return client.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+	installConfig := &installer.InstallConfig{}
+	if err := yaml.NewYAMLOrJSONDecoder(strings.NewReader(string(cm.Data["install-config"])), 100).Decode(installConfig); err != nil {
+		return nil, fmt.Errorf("unable to decode cluster install configuration: %v", err)
+	}
+
+	return installConfig, nil
 }
 
-func GetInfrastructure(kubeconfig *rest.Config) (*configv1.Infrastructure, error) {
-	client, err := configv1client.NewForConfig(kubeconfig)
+func GetInfrastructure(listers *regopclient.Listers) (*configv1.Infrastructure, error) {
```

Comment:
@bparees I think that the refactor of this method will help negate the issue with upgrades and PlatformStatus not being there, but ptal and let me know what you think.

Reply:
seems reasonable to me.  I didn't study the invocations and logic super closely, but in general it seems like a reasonable abstraction to protect yourself from the vagaries of 4.1 vs 4.2 install config apis.

Reply:
Thanks! That's all I was looking for.